tests/unit/lms/services/file_test.py

```diff
@@ -24,6 +25,33 @@ def test_it_doesnt_return_matching_files_from_other_application_instances(
 
         assert not svc.get(file_.lms_id, file_.type)
 
+    def test_it_upsert(self, db_session, svc, application_instance):
+        files = factories.File.create_batch(
+            5, application_instance=application_instance
+        )
+        db_session.flush()
+        total_files = db_session.query(File).count()
+
+        svc.upsert(
+            [
+                {
+                    "type": file.type,
+                    "course_id": file.course_id,
+                    "lms_id": file.lms_id,
+                    "name": f"file_{i}",
+                    "size": i,
+                }
+                for i, file in enumerate(files)
+            ]
+        )
+
+        assert db_session.query(File).count() == total_files
+        # Refresh the model instances to gather changed data from the DB
+        _ = [db_session.refresh(file) for file in files]
+        for i, file in enumerate(files):
+            assert file.size == i
+            assert file.name == f"file_{i}"
```

Comment:
Do we need a bit more testing here? Maybe some files that already exist (so they should get updated), some files that don't exist yet (should get created) (`factories.File.build_batch()` can create `models.File` objects without adding them to the DB), some files that already exist and already have the up-to-date values. It can probably still be all in one test